sdk/metric/instrument.go

```diff
@@ -214,3 +201,79 @@ func (i *instrumentImpl[N]) aggregate(ctx context.Context, val N, attrs []attrib
 		agg.Aggregate(val, attribute.NewSet(attrs...))
 	}
 }
+
+// observablID is a comparable unique identifier of an observable.
+type observablID[N int64 | float64] struct {
+	name        string
+	description string
+	kind        InstrumentKind
+	unit        unit.Unit
+	scope       instrumentation.Scope
+}
+
+type observable[N int64 | float64] struct {
+	instrument.Asynchronous
+	observablID[N]
+
+	aggregators []internal.Aggregator[N]
+}
+
+func newObservable[N int64 | float64](scope instrumentation.Scope, kind InstrumentKind, name, desc string, u unit.Unit, agg []internal.Aggregator[N]) *observable[N] {
+	return &observable[N]{
+		observablID: observablID[N]{
+			name:        name,
+			description: desc,
+			kind:        kind,
+			unit:        u,
+			scope:       scope,
+		},
+		aggregators: agg,
+	}
+}
+
+var _ instrument.Float64ObservableCounter = (*observable[float64])(nil)
+var _ instrument.Float64ObservableUpDownCounter = (*observable[float64])(nil)
+var _ instrument.Float64ObservableGauge = (*observable[float64])(nil)
+var _ instrument.Int64ObservableCounter = (*observable[int64])(nil)
+var _ instrument.Int64ObservableUpDownCounter = (*observable[int64])(nil)
+var _ instrument.Int64ObservableGauge = (*observable[int64])(nil)
+
+// Observe logs an error.
+func (o *observable[N]) Observe(ctx context.Context, val N, attrs ...attribute.KeyValue) {
+	var zero N
+	err := errors.New("invalid observation recording")
+	global.Error(err, "dropping measurement",
```

Comment:
Minor: 
Does this happen when an `Observe()` is made outside of the callback?  Can the error message be more helpful about what the correct behavior is?